client/client.go

```diff
@@ -361,11 +362,11 @@ func (c *authenticatingClient) createServiceURLs() error {
 	}
 	if len(missingServiceTypes) > 0 {
 		if len(possibleRegions) > 0 {
-			return fmt.Errorf("%s\none of these regions may be suitable instead: %s",
+			return gooseerrors.Newf(nil, "%s\none of these regions may be suitable instead: %s",
```

Comment:
I'm not sure that gooseerrors.Newf() is necessary in cases where we pass in nil.
The idea of Newf() is to add extra context to the error without losing the cause. This allows the various gooseerrors.IsFoo() methods to be used.
If we are passing in nil, then fmt.Errorf works just fine.